### DISCUSSION TITLE
How can I pass all response headers from my proxied httpd container to the browser?

### DISCUSSION THREAD

**Discussion by tngraessler** [nginx-proxy/nginx-proxy#1874] [Q&A] [answered]
Hi,

I use nginx-proxy for many web projects, most of them are PHP projects with Apache Webserver.
In these projects, I use .htaccess files to set the required response headers that should be read by the user's browser.
Unfortunately, these response headers don't show up when I open these websites in my browser.

Sure, I can create a custom per-virtual-host config file in the nginx-proxy container and use add_header to set these response headers.
BUT I'd rather keep those project data within the project and within the .htaccess file and I don't want to create an extra config file.

Therefore my question: **How can I pass all response headers from my proxied httpd container to the browser?**

Thanks and best regards,
Torsten

**Comment by tkw1536** [comment, marked as answer]
As far as I'm aware the proxy should pass all response headers automatically. 

If not, you might want to [replace the default proxy settings](https://github.com/nginx-proxy/nginx-proxy#replacing-default-proxy-settings) by mounting / creating `/etc/nginx/proxy.conf`.

It should be sufficient to add `proxy_pass_header <name>` for each header you want to pass (in addition to the defaults). Note that you don't need to add the header values into the nginx config, these can come from httpd (or whatever other application you are running).

**Reply by tngraessler** [reply]
Thank you!

I found my mistake: I haven't activated mod_headers in my proxied Apache Webserver. 🤦 Now, the response headers are correctly written to the output. 🥳

The only problem that I now is the duplicate output of the Strict-Transport-Security header. My application writes this header to the response and after that, the proxy also writes this response header – with another value. This results in a duplicated header in my browser:

strict-transport-security: max-age=31536000; includeSubDomains; preload;
strict-transport-security: max-age=31536000

I haven't figured out yet how to prevent the proxy from writing this response header.

**Reply by tkw1536** [reply]
Try adding `HSTS=off` to your container environment variables. That should completely turn off the built in hsts support, allowing you to do whatever you like manually. You might also want to play with other values. 

Do check out [the "How SSL Support Works" section of the README](https://github.com/nginx-proxy/nginx-proxy#how-ssl-support-works) for details.

**Reply by tngraessler** [reply]
This is the solution. Now it works as I want. Thank you very much! 😊
